packages/flutter_tools/lib/src/flutter_features_config.dart

```diff
@@ -0,0 +1,167 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'base/common.dart';
+import 'base/config.dart';
+import 'base/platform.dart';
+import 'features.dart';
+import 'flutter_manifest.dart';
+
+/// Reads configuration flags to possibly override the default flag value.
+///
+/// See [isEnabled] for details on how feature flag values are resolved.
+interface class FlutterFeaturesConfig {
+  /// Creates a feature configuration reader from the provided sources.
+  ///
+  /// [globalConfig] reads values stored by the `flutter config` tool, which
+  /// are normally in the user's `%HOME` directory (varies by system), while
+  /// [projectManfiest] reads values from the _current_ Flutter project's
+  /// `pubspec.yaml`
+  const FlutterFeaturesConfig({
+    required Config globalConfig,
+    required Platform platform,
+    required FlutterManifest? projectManifest,
+  }) : _globalConfig = globalConfig,
+       _platform = platform,
+       _projectManifest = projectManifest;
+
+  final Config _globalConfig;
+  final Platform _platform;
+
+  // Can be null if no manifest file exists in the current directory.
+  final FlutterManifest? _projectManifest;
+
+  /// Returns whether [feature] has been turned on/off from configuration.
+  ///
+  /// If the feature was not configured, or cannot be configured, returns `null`.
+  ///
+  /// The value is resolved, if possible, in the following order, where if a
+  /// step resolves to a boolean value, no further steps are attempted.
+  ///
+  /// ## Environment Variable
```

Comment:
Consider making it clearer this is an ordered list:

```suggestion
  /// The value is resolved, if possible, in the following order, where if a
  /// step resolves to a boolean value, no further steps are attempted:
  ///
  /// ## 1. Environment Variable
```